src/seedu/addressbook/Main.java

```diff
@@ -106,11 +106,14 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageFile.StorageReadOnlyException e) {
+            ui.showToUser(e.getMessage());
```

Comment:
Recall from tutorial on other ways we can improve on the way we handle exceptions!